File: HackerRank Solutions/30 Days of Code/Day 8 Dictionaries and Maps.py
# Enter your code here. Read input from STDIN. Print output to STDOUT
n = int(input())
phoneBook = dict()
for i in range(0, n):
    name, phoneNumber = input().split()
    phoneBook[name] = phoneNumber

for i in range(0, n):
    try:
        name = input()
        if name in phoneBook:
            print(f"{name}={phoneBook[name]}")
        else:
            print("Not found")
    except:
        break

File: HackerRank Solutions/30 Days of Code/Day 8 Dictionaries and Maps/Day 8 Dictionaries and Maps.cpp
#include <cmath>
#include <cstdio>
#include <vector>
#include <iostream>
#include <algorithm>
#include <map>
using namespace std;


int main() {
    /* Enter your code here. Read input from STDIN. Print output to STDOUT */
    int n;
    string name;
    long phoneNumber;
    cin >> n;
    cin.ignore();
    map <string,long> phoneBook;
    for(int i=0; i<n; i++){
        cin >> name;
        cin >> phoneNumber;
        phoneBook[name] = phoneNumber;
    }
    while(cin >> name){
        if(phoneBook.find(name) != phoneBook.end()){
            cout << name << "=" << phoneBook.find(name)->second << endl;
        } else{
            cout << "Not found" << endl;
        }
    }
    return 0;
}
